README.md

```diff
@@ -28,28 +19,52 @@ You get:
  * re(1)  — command line interface for executing regular expressions
  * lx(1)  — lexer generator
 
+lx is an attempt to produce a simple, expressive, and unobtrusive
+lexer generator which is good at lexing, does just lexing,
+is language independent, and has no other features.
+
 Clone with submodules (contains required .mk files):
 
     ; git clone --recursive https://github.com/katef/libfsm.git
 
-To build and install:
+### Building
 
-    ; pmake -r install
+To build and install:
 
-To build on Linux, you may need to install `bmake`, and use it instead
-of `pmake` in the commands mentioned here.
+    ; bmake -r
+    ; bmake -r install
 
 You can override a few things:
 
-    ; CC=clang PREFIX=$HOME pmake -r install
+    ; CC=clang bmake -r
+    ; PREFIX=$HOME bmake -r install
 
-Building depends on:
+To run the tests:
 
- * Any BSD make. This includes OpenBSD, FreeBSD and NetBSD make(1)
-   and sjg's portable bmake (also packaged as pmake).
+    ; bmake -r LX=./build/bin/lx test
 
- * A C compiler. Any should do, but GCC and clang are best supported.
+You need bmake for building.
+
+ * If you use some kind of BSD (NetBSD, OpenBSD, FreeBSD, ...) this is make(1).
+   They all differ slightly. Any of them should work.
+ * If you use Linux or MacOS and you have a package named bmake, use that.
+ * If you use Linux and you have a package named pmake, use that.
+   It's the same thing.
+   Some package managers have bmake packaged under the name pmake.
+   I don't know why they name it pmake.
+ * Otherwise if you use MacOS and you only have a package named bsdmake, use that.
+   It's Apple's own fork of bmake.
+   It should also work but it's harder for me to test.
+ * If none of these are options for you, you can build bmake from source.
+   You don't need mk.tar.gz, just bmake.tar.gz.
+   https://www.crufty.net/help/sjg/bmake.html
```

Comment:
I like the detail here; there's a lot.  I was initially worried about it being confusing, but re-reading it, I don't think it is.

One thing you may clarify is that sjg's `bmake` always works (as long as it compiles).  I've run into problems with `bmake/pmake/bsdmake` on Linux and macOS before.

Reply:
Added in 2bee00a, and I numbered the list items.